src/libswid.cpp

```diff
@@ -76,20 +76,27 @@ unsigned int swid_root_get_type(SWIDHandle data) {
  */
 
 SWIDIOHandle swid_create_io(const char * backend) {
-	return (SWIDIOHandle)get_swidtagio(backend);
+	SWIDIOHandle io = new SWIDTagIO();
```

Comment:
hiding the fact that the handle is a pointer may not be a good idea, people will probably not remember to free it

Reply:
I think that this is OK. It is documented, it is in the example, and a red light turns on in every C programmer's head whenever one calls a function containing `create`.